lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -94,39 +98,57 @@ private void validateFeature(DOMDocument domDocument, List<Diagnostic> list, DOM
         }
     }
 
-    private void monitorConfigFiles(DOMDocument domDocument, List<Diagnostic> list, DOMNode node) {
+    /**
+     * Location checks:
+     * 1) Relative path; 2) Server config dir; 
+     * 3) Absolute path; 4) Web server
+     * 
+     * Checks in method: 1), 3)
+     * 2) performed in isConfigXMLFile
+     * 4) not yet implemented/determined
+     */
+    private void validateIncludeLocation(DOMDocument domDocument, List<Diagnostic> list, DOMNode node) {
         String locAttribute = node.getAttribute("location");
         if (locAttribute == null) {
             return;
         }
+        // skip diagnostic for not yet implemented behaviors/checks (URL + vars)
+        if (locAttribute.startsWith("http") || locAttribute.contains("$")) {
+            return;
+        }
 
         DOMNode locNode = node.getAttributeNode("location");
         Range range = XMLPositionUtility.createRange(locNode.getStart(), locNode.getEnd(), domDocument);
-
         if (!locAttribute.endsWith(".xml")) {
-            String message = "The specified file is not an XML file.";
+            String message = "The specified resource is not an XML file.";
             list.add(new Diagnostic(range, message, DiagnosticSeverity.Warning, "liberty-lemminx"));
             return;
         }
 
-        String docURIString = domDocument.getDocumentURI().replace(File.separator, "/");
-        locAttribute = locAttribute.replace(File.separator, "/");
-        File configFile = locAttribute.startsWith("./") ? 
-                new File(URI.create(docURIString.substring(0, docURIString.lastIndexOf("/") + 1))
-                        .resolve(locAttribute).normalize()) :
-                new File(locAttribute);
-
+        String optString = node.getAttribute("optional");
+        boolean required = optString == null ? false : optString.equals("false");
 
+        File docParentFile = LibertyUtils.getDocumentAsFile(domDocument).getParentFile();
+        File configFile = new File(docParentFile, locAttribute);    
+        if (!configFile.exists()) {
+            configFile = new File(locAttribute);
+        }
         try {
             if (configFile.exists()) {
-                LibertyProjectsManager.getInstance().getWorkspaceFolder(docURIString).addConfigFile(configFile.getCanonicalPath());
+                LibertyProjectsManager.getInstance().getWorkspaceFolder(domDocument.getDocumentURI()).addConfigFile(configFile.getCanonicalPath());
             } else {
-                String message = "The file at the specified location could not be found.";
-                list.add(new Diagnostic(range, message, DiagnosticSeverity.Warning, "liberty-lemminx"));
+                if (required) {
```

Comment:
You said the optional flag defaults to `false` in another comment. So if it is null above at line 129, shouldn't `required` default to `true`?